Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/transport/round_trippers.go

```diff
@@ -130,6 +133,36 @@ func (rt *basicAuthRoundTripper) CancelRequest(req *http.Request) {
 
 func (rt *basicAuthRoundTripper) WrappedRoundTripper() http.RoundTripper { return rt.rt }
 
+type actAsRoundTripper struct {
+	actAs string
+	rt    http.RoundTripper
+}
+
+// NewActAsRoundTripper will apply a BASIC auth authorization header to a
+// request unless it has already been set.
+func NewActAsRoundTripper(actAs string, rt http.RoundTripper) http.RoundTripper {
+	return &actAsRoundTripper{actAs, rt}
+}
+
+func (rt *actAsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if len(req.Header.Get("Act-As")) != 0 {
+		return rt.rt.RoundTrip(req)
+	}
+	req = cloneRequest(req)
+	req.Header.Set("Acting-As", rt.actAs)
```

Comment:
@lavalamp @bgrant0607 do you have a preference on what prefix to use for kubernetes-api-specific headers?

@deads2k @smarterclayton do you mind an `X-Kubernetes-*` or `X-Kube-*` or `X-K8s-*` prefix for http headers that are common to kubernetes and Openshift? 

The reason this is a header is that (1) it is for authentication, which already uses headers, and (2), it applies to almost all requests, so adding a query parameter to all resources and methods would be pretty verbose.